python/tests/unit/connectors/memory/azure_cosmos_db/test_azure_cosmos_db_mongodb_collection.py

```diff
@@ -0,0 +1,292 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+from pydantic import BaseModel, ValidationError
+from pydantic_core import InitErrorDetails
+from pymongo import AsyncMongoClient
+
+import semantic_kernel.connectors.memory.azure_cosmos_db.azure_cosmos_db_mongodb_collection as cosmos_collection
+import semantic_kernel.connectors.memory.azure_cosmos_db.azure_cosmos_db_mongodb_settings as cosmos_settings
+from semantic_kernel.data.const import DistanceFunction, IndexKind
+from semantic_kernel.data.record_definition import (
+    VectorStoreRecordDataField,
+    VectorStoreRecordDefinition,
+    VectorStoreRecordKeyField,
+    VectorStoreRecordVectorField,
+)
+from semantic_kernel.exceptions import VectorStoreInitializationException
+
+
+class FakeDataModel:
+    """
+    A fake data model class to satisfy the "type" requirement for data_model_type.
+    """
+
+    pass
+
+
+async def test_constructor_with_mongo_client_provided() -> None:
+    """
+    Test the constructor of AzureCosmosDBforMongoDBCollection when a mongo_client
+    is directly provided. Expect that the class is successfully initialized
+    and doesn't attempt to manage the client.
+    """
+    mock_client = AsyncMock(spec=AsyncMongoClient)
+    collection_name = "test_collection"
+    fake_definition = VectorStoreRecordDefinition(
+        fields={
+            "id": VectorStoreRecordKeyField(),
+            "content": VectorStoreRecordDataField(),
+            "vector": VectorStoreRecordVectorField(),
+        }
+    )
+
+    collection = cosmos_collection.AzureCosmosDBforMongoDBCollection(
+        collection_name=collection_name,
+        data_model_type=FakeDataModel,
```

Comment:
I usually test with a dict instead, when there is a definition only
```suggestion
        data_model_type=dict,
```